index.js

```diff
@@ -69,30 +67,18 @@ module.exports = class ServeDrive extends ReadyResource {
     return this.server.address()
   }
 
-  getLink (path, id, opts = {}) {
-    if (id && typeof id !== 'string') return this.getLink(path, null, id)
-    if (typeof opts === 'number') opts = { version: opts }
-
-    const version = opts.version
-    const protocol = opts.protocol || 'http'
-    const domain = opts.domain
-    const port = opts.port
-
-    path = unixPathResolve('/', path)
-
-    let link = `${protocol}://`
-
-    link += domain
-      ? domain + (port ? `:${port}` : '')
-      : `${LOCALHOST}:${this.address().port}`
-    link += path
+  getLink (path, opts = {}) {
+    const pathname = unixPathResolve('/', path)
+    const proto = opts.secure ? 'https' : 'http'
+    const host = opts.host || '127.0.0.1'
```

Comment:
Bit of nomenclature confusion here, because we're using this opt to specify either an ip address or a domain name. Can't think of a cleaner name though, so fine with host

Reply:
Yeah, that might be a good solution too: either you don't specify a host, and it uses the server's address+port, or you do and you have to specify it completely. I think that covers all use cases and avoids the complexity